src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl<T extends Machine> implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        MachineProducer machineProducer;
+        List<Machine> list = new ArrayList<>();
+        if (Bulldozer.class.equals(type)) {
+            machineProducer = new BulldozerProducer();
+            list = new ArrayList<>(machineProducer.get());
+        } else if (Excavator.class.equals(type)) {
+            machineProducer = new ExcavatorProducer();
+            list = new ArrayList(machineProducer.get());
+        } else if (Truck.class.equals(type)) {
+            machineProducer = new TruckProducer();
+            list = new ArrayList<>(machineProducer.get());
+        }
+        return list;
```

Comment:
This method must be a good example of polymorphism. Let's call method `get` only once